src/unit_tests/remoted/test_sendmsg.c

```diff
@@ -164,10 +298,313 @@ void test_send_msg_tcp_err(void ** state) {
     assert_int_equal(remoted_state.queued_msgs, 0);
 }
 
+void test_send_msg_tcp_err_closed_socket(void ** state) {
+    (void) state;
+
+    char *agent_id = "001";
+    char *msg = "abcdefghijk";
+    ssize_t msg_length = 11;
+    int key = 0;
+
+    char *crypto_msg = "!@#123abc";
+    ssize_t crypto_size = 9;
```

Comment:
Hi @jftuduri

Same change here

```suggestion
    ssize_t crypto_size = strlen(msg);
```

Reply:
Fixed